tests/bluetooth/shell/src/main.c

```diff
@@ -2052,12 +2077,7 @@ static void l2cap_disconnected(struct bt_l2cap_chan *chan)
 	printk("Channel %p disconnected\n", chan);
 }
 
-static struct net_buf *l2cap_alloc_buf(struct bt_l2cap_chan *chan)
-{
-	printk("Channel %p requires buffer\n", chan);
-
-	return net_buf_alloc(&data_rx_pool, K_FOREVER);
-}
+static struct net_buf *l2cap_alloc_buf(struct bt_l2cap_chan *chan);
```

Comment:
Don't think that's possible since it accesses l2cap_ops which is declared below. So the forward declaration is unavoidable due to the circular dependency.